FeatureExtraction/Shape/extract_shape_features.py

```diff
@@ -0,0 +1,173 @@
+from collections import OrderedDict
+import numpy as np
+from skimage.measure import regionprops, label
+from data_types import SegmentedImage, GreyscaleImage
+
+
+def extract_shape_features(segmented_image: SegmentedImage,
+                           minimal_cell_size: int = 50) -> OrderedDict:
+    """
+    Function for extracting shape features from a image with segmented blue nuclei.
+    :param segmented_image: Image with segmented blue nuclei.
+    :param minimal_cell_size: Minimal are of cell to be counted
+    :return: Ordered dict with features:
+    number_of_cells;
+    total area of cells;
+    coefficient of total are of cells and total area of an image;
+    min, mean, max, std of area;
+    min, mean, max, std of perimeter;
+    min, mean, max, std of eccentricity;
+    min, mean, max, std of solidity;
+    """
+    if not isinstance(segmented_image, SegmentedImage):
+        raise TypeError(
+            "Image should be segmentation, use SegmentedImage type")
+
+    segmented_image = GreyscaleImage(segmented_image)
+    mask = segmented_image[...] != 0
+    masked = segmented_image.copy()
+    masked[mask] = 255
+    labels = label(masked)
+
+    props = regionprops(labels)
+    number_of_cells = count_cells(props, minimal_cell_size)
+    try:
```

Comment:
Maybe this `try` segment would be more fitting in the examined function itself? Same for others.